src/uu/cp/src/cp.rs

```diff
@@ -1086,6 +1086,38 @@ fn copy_directory(
         .into());
     }
 
+    // If in `--parents` mode, create all the necessary ancestor directories.
+    //
+    // For example, if the command is `cp --parents a/b/c d`, that
+    // means we need to copy the two ancestor directories first:
+    //
+    // a -> d/a
+    // a/b -> d/a/b
+    //
+    let tmp = if options.parents {
+        if let Some(parent) = root.parent() {
+            let new_target = target.join(parent);
+            std::fs::create_dir_all(&new_target)?;
+
+            if options.verbose {
+                let mut ancestors = vec![];
+                for p in parent.ancestors() {
+                    ancestors.push(p);
+                }
+                for p in ancestors.iter().rev().skip(1) {
+                    println!("{} -> {}", p.display(), target.join(p).display());
```

Comment:
The ancestors are printed without quotes, so using `context_for()` is not appropriate here.